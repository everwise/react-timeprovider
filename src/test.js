import * as React from 'react';
import TestRenderer from 'react-test-renderer';

import { createTimeProvider, GetTime, withTime } from './';

const MockTimeProvider = createTimeProvider(() => '2018–04–06T12:30:00Z');

describe('TimeProvider', () => {
  test('GetTime can get the time from the Provider', () => {
    const tree = TestRenderer.create(
      <MockTimeProvider>
        <GetTime>{({ currentTime }) => currentTime}</GetTime>
      </MockTimeProvider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  const TimeOutputter = withTime(({ currentTime }) => <span>{currentTime}</span>);

  test('withTime can get the time from the Provider', () => {
    const tree = TestRenderer.create(
      <MockTimeProvider>
        <TimeOutputter />
      </MockTimeProvider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
