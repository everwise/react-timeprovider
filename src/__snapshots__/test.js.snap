// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeProvider GetTime can get the time from the Provider 1`] = `"2018–04–06T12:30:00Z"`;

exports[`TimeProvider withTime can get the time from the Provider 1`] = `
<span>
  2018–04–06T12:30:00Z
</span>
`;
